app/src/main/java/protect/card_locker/CatimaRemoteViewsService.kt

```diff
@@ -0,0 +1,132 @@
+        val sortPref: SharedPreferences = context.getSharedPreferences(
+            "sharedpreference_sort",
+            Context.MODE_PRIVATE
+        )
+
+        val orderString = sortPref.getString("sharedpreference_sort_order", null)
+        val orderDirectionString =
+            sortPref.getString("sharedpreference_sort_direction", null)
+
+        if (orderString != null && orderDirectionString != null) {
+            try {
+                mOrder = DBHelper.LoyaltyCardOrder.valueOf(orderString)
+                mOrderDirection = DBHelper.LoyaltyCardOrderDirection.valueOf(orderDirectionString)
+            } catch (ignored: IllegalArgumentException) {
+            }
+        }
+
+        val cur = DBHelper.getLoyaltyCardCursor(
+            mDatabase,
+            "",
+            null,
+            mOrder,
+            mOrderDirection,
+            LoyaltyCardArchiveFilter.All
+        )
+
+        mCards.clear()
+        if (cur.moveToFirst()) {
+            do {
+                val item = LoyaltyCard.fromCursor(context, cur)
+                mCards.add(item)
+            } while (cur.moveToNext())
+        }
+    }
+
+    override fun onDestroy() {}
+
+    override fun getCount(): Int {
+        return mCards.count()
+    }
+
+    @RequiresApi(23)
+    private fun createRemoteView(item: LoyaltyCard): RemoteViews
+    {
+        val rv = RemoteViews(context.packageName, R.layout.catima_widget_item
+        ).apply {
+            val backColor = Utils.getHeaderColor(context, item)
+            setInt(R.id.item_container, "setBackgroundColor", backColor)
+
+            val icon = item.getImageThumbnail(context)
+            if (icon != null) {
+                Log.e(
+                    "RV",
+                    String.format("Icon %s size: %dx%d %d", item.store, icon.width, icon.height, icon.byteCount)
+                )
+                //setImageViewBitmap(R.id.item_image, icon)   //not working?? image not changing after reorder cards!!
+                setImageViewIcon(R.id.item_image, Icon.createWithBitmap(icon))   //working?? image is changing every time ok !!
+                setViewVisibility(R.id.item_text, View.INVISIBLE)
+                setViewVisibility(R.id.item_image, View.VISIBLE)
+            } else {
+                setImageViewBitmap(R.id.item_image, null)
+                setTextViewText(R.id.item_text, item.store)
+                setViewVisibility(R.id.item_text, View.VISIBLE)
+                setViewVisibility(R.id.item_image, View.INVISIBLE)
+                setTextColor(
+                    R.id.item_text,
+                    if (Utils.needsDarkForeground(backColor)) Color.BLACK else Color.WHITE
+                )
+            }
+
+            val fillInIntent = Intent().apply {
+                putExtra("item_id", item.id)
+            }
+
+            setOnClickFillInIntent(R.id.item_container, fillInIntent)
+        }
+
+        return rv
+    }
+
+    @RequiresApi(Build.VERSION_CODES.M)
```

Comment:
Also here, I'll need to check Android 5.x behaviour. But feel free to make that my problem :P